android/src/main/java/com/reactnativestripesdk/pushprovisioning/AddToWalletButtonView.kt

```diff
@@ -0,0 +1,152 @@
+package com.reactnativestripesdk.pushprovisioning
+
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.RippleDrawable
+import android.view.MotionEvent
+import androidx.appcompat.widget.AppCompatImageView
+import com.bumptech.glide.RequestManager
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.load.model.GlideUrl
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.uimanager.events.EventDispatcher
+import com.reactnativestripesdk.PushProvisioningProxy
+import com.reactnativestripesdk.createError
+
+
+class AddToWalletButtonView(private val context: ThemedReactContext, private val requestManager: RequestManager) : AppCompatImageView(context) {
+  private var cardDescription: String? = null
+  private var cardLastFour: String? = null
+  private var ephemeralKey: String? = null
+  private var sourceMap: ReadableMap? = null
+  private var token: ReadableMap? = null
+
+  private var mEventDispatcher: EventDispatcher? = context.getNativeModule(UIManagerModule::class.java)?.eventDispatcher
+  private var loadedSource: GlideUrl? = null
+  private var heightOverride: Int = 0
+  private var widthOverride: Int = 0
+
+  override fun performClick(): Boolean {
```

Comment:
While testing performClick without push provisioning dependency, I see a few unhandled promise rejections. Should these be handled?

```
(node:88963) UnhandledPromiseRejectionWarning: Error: Passing apiVersion in a separate options hash is required to create an ephemeral key. See https://stripe.com/docs/api/versioning?lang=node
    at Object.validator (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/resources/EphemeralKeys.js:16:15)
    at getRequestOpts (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/makeRequest.js:59:10)
    at /Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/makeRequest.js:83:14
    at new Promise (<anonymous>)
    at makeRequest (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/makeRequest.js:80:10)
    at Constructor.create (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/StripeMethod.js:38:7)
    at _callee7$ (/Users/jameswoo/stripe/stripe-react-native/example/dist/index.js:1:14451)
    at tryCatch (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/regenerator-runtime/runtime.js:63:40)
    at Generator.invoke [as _invoke] (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/regenerator-runtime/runtime.js:294:22)
    at Generator.next (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/regenerator-runtime/runtime.js:119:21)
(node:88963) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 11)
(node:88963) UnhandledPromiseRejectionWarning: Error: No such issuing card: 'ic_1KnTM2F05jLespP6wNLZQ1mu'
    at Function.generate (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/Error.js:40:16)
    at res.toJSON.then.StripeAPIError.message (/Users/jameswoo/stripe/stripe-react-native/example/node_modules/stripe/lib/StripeResource.js:220:35)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(node:88963) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 12)
```

Reply:
This is a rejection coming from the client app code, not our SDK code. Your getting an empty string value for [this constants value](https://github.com/stripe/stripe-react-native/blob/charliecruzan-push-provisioning/example/src/screens/GooglePayScreen.tsx#L157) so the server is complaining when you try to fetch an ephemeral key.

Reply:
so basically- it's expected